### DISCUSSION TITLE
Runnung MagicScaler API in .net core middleware

### DISCUSSION THREAD

**Discussion by Astatira** [saucecontrol/PhotoSauce#99] [Q&A] [answered]
Hi,

Thanks for the great library!
I wanna use your library to resize images beased on query string. I don't want to save the new image on hard disk  based on new settings, just want to response on the fly to the client side. how can I make a middleware for this in .net core ? 

In fact I need somthing like this answer (User711641945 posted) for SkiaSharp in this link 
[https://social.msdn.microsoft.com/Forums/en-US/8d5eb833-6911-4208-aadc-974c7c7da2a9/image-resize-and-crop?forum=aspdotnetcore](url)
but implemented by your library?

**Comment by saucecontrol** [comment, marked as answer]
Howdy!

You shouldn't have any issue at all using MagicScaler in a middleware component.  You don't need to save to a file, just pass any writeable Stream (e.g. `MemoryStream`, or even better, [RecyclableMemoryStream](https://github.com/microsoft/Microsoft.IO.RecyclableMemoryStream)) for the output parameter to `ProcessImage`.

The [tutorial](https://www.paddo.org/asp-net-core-image-resizing-middleware/) mentioned in that MSDN social post should have everything you need to get started.  You can ignore all the Skia code and particularly the workarounds for Skia quirks and limitations, as MagicScaler will do the correct thing by default when you use `ProcessImage`.

**Reply by Astatira** [reply]
Thnx for your help and seggestion of RecyclableMemoryStream . It works perfectlly. 
But we've faced a minor issue with the default setting, we can't upscale the image, and the maximum image output is the original dimensions! 
the same behavior even with " CropScaleMode.Contain ".
All works perfectly, with downscale and cropping but upscale.
is there any option to enable the upscaling behavior?!
thanks in advance.

**Reply by saucecontrol** [reply]
Upscale is allowed in all `CropScaleMode`s except `Max`.  Can you share code you're using that does otherwise?

**Reply by Astatira** [reply]
Ooops, my bad! We had a typo. After your confirmation we double checked all and now all work perfectly as expected.
Thank you again for your great work and prompt reply. 👍 we fell in love with your library!
